pkg/ratelimiter/ratelimiter.go

```diff
@@ -0,0 +1,81 @@
+package ratelimiter
+
+import (
+	"time"
+
+	"github.com/k8stopologyawareschedwg/resource-topology-exporter/pkg/notification"
+	"go.uber.org/ratelimit"
+)
+
+type RateLimitedEventSource struct {
+	es    notification.EventSource
+	inCh  <-chan notification.Event
+	outCh chan notification.Event
+	rt    ratelimit.Limiter
+
+	doneChan chan struct{}
+	stopChan chan struct{}
+}
+
+func NewRateLimitedEventSource(es notification.EventSource, maxEventsPerTimeUnit uint64, timeUnit time.Duration) (*RateLimitedEventSource, error) {
+
+	rles := RateLimitedEventSource{
+		es:       es,
+		inCh:     es.Events(),
+		outCh:    make(chan notification.Event),
+		doneChan: make(chan struct{}, 1),
+		stopChan: make(chan struct{}),
+	}
+
+	options := ratelimit.Per(timeUnit)
+	rles.rt = ratelimit.New(int(maxEventsPerTimeUnit), options)
```

Comment:
let's say we get a burst of 100 events in the time unit, while we are configured to emit only 1 event per time unit. Which event among the initial 100 is going to be emitted? the first (aka the oldest)? the last (aka the newest)? something else?

Reply:
None of them. 
I mean, my first approach in #85 was to use a buffered channel as input channel and use a select in the write operation to that channel so when the buffered channel went full the new writes will be dropped, that would make the input and output rates independent. 
But in the review we thought it was better to use a blocking channel. In this way the writer will be blocked in the channel until the reader gets the value, so no burst will occur because the reader will make the writer to go slower.
I still prefer the first approach.